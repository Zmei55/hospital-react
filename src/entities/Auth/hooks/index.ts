export * from "./useLogIn";
export * from "./useLogOut";
export * from "./useHandleChange";
