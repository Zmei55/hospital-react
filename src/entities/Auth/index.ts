export * from "./components";
export * from "./hooks";
export * from "./redux";
export * from "./types";
export * from "./data";
