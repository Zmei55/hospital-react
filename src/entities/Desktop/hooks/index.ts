export * from "./useNewRequest";
export * from "./useFindRequest";
