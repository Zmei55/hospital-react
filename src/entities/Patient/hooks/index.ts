export * from "./useChoosePatient";
export * from "./useGetPatient";
export * from "./usePatientsListFind";
export * from "./useHandleInfoActive";
export * from "./useClearPatient";
export * from "./useTogglePatientsModal";
