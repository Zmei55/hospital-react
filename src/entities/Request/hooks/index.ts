export * from "./useRequestServices";
export * from "./useSendRequestDetails";
