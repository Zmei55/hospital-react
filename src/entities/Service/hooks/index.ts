export * from "./useServiceAdd";
export * from "./useServiceDelete";
export * from "./useSelectedServiceDelete";
export * from "./useSelectedServiceListSave";
export * from "./useSelectedServiceListClear";
export * from "./useGetLaborsList";
export * from "./useGetServiceById";
export * from "./useGetServiceListById";
export * from "./useHandleLaborChange";
export * from "./useHandleServicesForm";
export * from "./useToggleLaborModal";
export * from "./useToggleServicesModal";
