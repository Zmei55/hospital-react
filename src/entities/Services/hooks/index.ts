export * from "./useSavedServicesList";
export * from "./useServicesFilter";
export * from "./useAddService";
