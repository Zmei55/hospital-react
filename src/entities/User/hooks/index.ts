export * from "./useGetUserData";
export * from "./useToggleMenuModal";
