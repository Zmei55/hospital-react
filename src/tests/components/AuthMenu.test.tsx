import { render, screen } from "@testing-library/react";
import { allTheProviders } from "tests/utils";
import userEvent from "@testing-library/user-event";
import { AuthMenu } from "entities/Auth";
import * as authHooks from "entities/Auth/hooks";

jest.mock("entities/Auth/hooks");

const mockedUseLogIn = jest.spyOn(authHooks, "useLogIn");

describe("AuthMenu component", () => {
  it("AuthMenu renders", () => {
    mockedUseLogIn.mockReturnValue({
      isLoading: false,
      isError: false,
      handleSignInSubmit: jest.fn(),
      errorModalClose: jest.fn(),
    });

    const { container } = render(allTheProviders(<AuthMenu />));

    expect(screen.getByTestId("auth-page")).toBeInTheDocument();
    expect(screen.getByText(/authorization/i)).toBeInTheDocument();
    expect(screen.getByText(/enter login and password/i)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("inputs renders", () => {
    mockedUseLogIn.mockReturnValue({
      isLoading: false,
      isError: false,
      handleSignInSubmit: jest.fn(),
      errorModalClose: jest.fn(),
    });

    render(allTheProviders(<AuthMenu />));

    expect(screen.getByRole("textbox")).toBeInTheDocument(); // нашли инпут по роли
    expect(screen.getByPlaceholderText("Username")).toBeInTheDocument(); // нашли инпут с Placeholder "Username"
    expect(screen.getByPlaceholderText("Password")).toBeInTheDocument(); // нашли инпут с Placeholder "Password"
  });

  it("user sign-in", async () => {
    mockedUseLogIn.mockReturnValue({
      isLoading: false,
      isError: false,
      handleSignInSubmit: jest.fn(),
      errorModalClose: jest.fn(),
    });

    render(allTheProviders(<AuthMenu />));

    const signInBtn = screen.getByTestId("sign-in-btn"); // кнопка входа
    expect(signInBtn).toBeInTheDocument(); // подтвердили наличие кнопки входа

    await userEvent.type(screen.getByPlaceholderText("Username"), "test"); // написали в инпут "test"
    expect(screen.getByPlaceholderText("Username")).toHaveValue("test"); // подтвердили значение инпут "test"
    await userEvent.type(screen.getByTestId("auth-password-input"), "test123");
    expect(screen.getByTestId("auth-password-input")).toHaveValue("test123");

    userEvent.click(signInBtn);
  });

  it("AuthMenu ErrorModal renders", async () => {
    mockedUseLogIn.mockReturnValue({
      isLoading: false,
      isError: true,
      handleSignInSubmit: jest.fn(),
      errorModalClose: jest.fn(),
    });

    const { container } = render(allTheProviders(<AuthMenu />));

    expect(screen.getByTestId("backdrop")).toBeInTheDocument();
    expect(screen.getByTestId("modal-content")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("Login button spinner renders", () => {
    mockedUseLogIn.mockReturnValue({
      isLoading: true,
      isError: false,
      handleSignInSubmit: jest.fn(),
      errorModalClose: jest.fn(),
    });

    render(allTheProviders(<AuthMenu />));

    expect(screen.getByTestId("spinner")).toBeInTheDocument();
  });
});
