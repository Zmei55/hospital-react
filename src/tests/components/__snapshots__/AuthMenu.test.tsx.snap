// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthMenu component AuthMenu ErrorModal renders 1`] = `
<div>
  <div
    class="css-1asre80"
    data-testid="auth-page"
  >
    <form
      class="css-1oskuwq"
      data-testid="auth-form"
    >
      <h2
        class="css-vb98sm"
        data-testid="auth-title"
      >
        Authorization
      </h2>
      <h3
        class="css-ja89o5"
        data-testid="auth-subtitle"
      >
        Enter login and password
      </h3>
      <div
        class="css-1nbhxtp"
        data-testid="auth-inputs-container"
      >
        <label
          class="css-1oha7z1"
          data-testid="auth-username-label"
        >
          <input
            class="css-1ry88mf"
            data-testid="auth-username-input"
            name="username"
            placeholder="Username"
            required=""
            style="width: 368px;"
          />
        </label>
        <label
          class="css-1oha7z1"
          data-testid="auth-password-label"
        >
          <input
            class="css-1ry88mf"
            data-testid="auth-password-input"
            name="password"
            placeholder="Password"
            required=""
            style="width: 368px;"
            type="password"
          />
        </label>
      </div>
      <button
        class="css-mbrcjd"
        data-testid="sign-in-btn"
        disabled=""
        style="width: 368px; height: 72px;"
        type="submit"
      >
        Login
        <svg
          class="css-bvj43j"
          fill="white"
          height="24"
          width="24"
        >
          <use
            href="svgSprite.svg#caret-right-bold"
          />
        </svg>
      </button>
    </form>
  </div>
</div>
`;

exports[`AuthMenu component AuthMenu renders 1`] = `
<div>
  <div
    class="css-1asre80"
    data-testid="auth-page"
  >
    <form
      class="css-1oskuwq"
      data-testid="auth-form"
    >
      <h2
        class="css-vb98sm"
        data-testid="auth-title"
      >
        Authorization
      </h2>
      <h3
        class="css-ja89o5"
        data-testid="auth-subtitle"
      >
        Enter login and password
      </h3>
      <div
        class="css-1nbhxtp"
        data-testid="auth-inputs-container"
      >
        <label
          class="css-1oha7z1"
          data-testid="auth-username-label"
        >
          <input
            class="css-1ry88mf"
            data-testid="auth-username-input"
            name="username"
            placeholder="Username"
            required=""
            style="width: 368px;"
          />
        </label>
        <label
          class="css-1oha7z1"
          data-testid="auth-password-label"
        >
          <input
            class="css-1ry88mf"
            data-testid="auth-password-input"
            name="password"
            placeholder="Password"
            required=""
            style="width: 368px;"
            type="password"
          />
        </label>
      </div>
      <button
        class="css-mbrcjd"
        data-testid="sign-in-btn"
        disabled=""
        style="width: 368px; height: 72px;"
        type="submit"
      >
        Login
        <svg
          class="css-bvj43j"
          fill="white"
          height="24"
          width="24"
        >
          <use
            href="svgSprite.svg#caret-right-bold"
          />
        </svg>
      </button>
    </form>
  </div>
</div>
`;
